x-pack/solutions/observability/plugins/slo/server/services/tasks/temp_summary_cleanup_task.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { errors } from '@elastic/elasticsearch';
+import { type CoreSetup, type Logger, type LoggerFactory } from '@kbn/core/server';
+import {
+  ConcreteTaskInstance,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '@kbn/task-manager-plugin/server';
+import { getDeleteTaskRunResult } from '@kbn/task-manager-plugin/server/task';
+import { SLOConfig } from '../../types';
+import { CleanUpTempSummary } from '../management/clean_up_temp_summary';
+
+export const TYPE = 'slo:temp-summary-cleanup-task';
+export const VERSION = '1.0.0';
+
+interface TaskSetupContract {
+  taskManager: TaskManagerSetupContract;
+  core: CoreSetup;
+  logFactory: LoggerFactory;
+  config: SLOConfig;
+}
+
+export class TempSummaryCleanupTask {
+  private abortController = new AbortController();
+  private logger: Logger;
+  private config: SLOConfig;
+  private wasStarted: boolean = false;
+
+  constructor(setupContract: TaskSetupContract) {
+    const { core, config, taskManager, logFactory } = setupContract;
+    this.logger = logFactory.get(this.taskId);
+    this.config = config;
+
+    taskManager.registerTaskDefinitions({
+      [TYPE]: {
+        title: 'SLO temp summary cleanup task',
+        timeout: '5m',
+        maxAttempts: 1,
+        createTaskRunner: ({ taskInstance }: { taskInstance: ConcreteTaskInstance }) => {
+          return {
+            run: async () => {
+              return this.runTask(taskInstance, core);
+            },
+
+            cancel: async () => {
+              this.abortController.abort('Timed out');
+            },
+          };
+        },
+      },
+    });
+  }
+
+  public async start({ taskManager }: { taskManager: TaskManagerStartContract }) {
+    if (!taskManager) {
+      this.logger.error('Missing required service during start');
+      return;
+    }
+
+    if (!this.config.tempSummaryCleanupTaskEnabled) {
+      return await taskManager.removeIfExists(this.taskId);
+    }
+
+    this.logger.info(`Started with 1h interval`);
+    this.wasStarted = true;
+
+    try {
+      await taskManager.ensureScheduled({
+        id: this.taskId,
+        taskType: TYPE,
+        scope: ['observability', 'slo'],
+        schedule: {
+          interval: '1h',
```

Comment:
Do you think it makes sense to further optimise this task? i imagine this is only needed after new SLO is created, so let's say when task runs, and it doesn't find any temporary documents you can increase it's interval? 

and possible in the path where SLO is created , you can reschedule it again with more urgent frequency.

Reply:
I've tested the search query on QA where there are 1M+ instances, and it returns in 286ms. But if that's a concern, I could increase the interval to a day? 

![image](https://github.com/user-attachments/assets/7ed38e00-a57f-4ddf-85fb-d9c6e7468ef7)


It is needed after creation, reset and breaking updates so we would need to schedule the task after each one of these actions. We would also need to add a search query that would check the number of temporary documents so we could unschedule the task in case there is none.

Reply:
I discussed with @ymao1 about either managing the scheduling ourselves or keeping it scheduled, and it is fine to have the task continuously scheduled since we would return early in case there is nothing to do. 

Quoting her: "i think in that case, there's no harm in having the task scheduled to run and short circuit early if there's nothing to do. It shouldn't clog up the task queue and then you won't have to manage the scheduling logic yourself"

I'm suggesting to add an even faster short circuit condition by querying the `summary-temp` index directly, and if there is no hits, return early. @shahzad31 would that work for you?

Reply:
that's fine by me, i always be careful when adding a new task , since kibana only supports around 200 task per minute in a normal deployment so if every team adds 2-4 tasks for cleaning things it can adds up to a lot. 

We already have an orphan summary clean up task that also runs every hour so maybe you can combine both of these things in same tasks. But it's an opinionated choice. Otherwise short-circuiting will greatly help. 

Thank you providing detailed feedback on my quesiton.